src/test/java/org/mockito/internal/util/DefaultMockingDetailsTest.java

```diff
@@ -42,28 +44,51 @@ public void should_check_that_a_spy_is_also_a_mock() throws Exception {
     }
 
     @Test
-    public void should_get_mocked_type() throws Exception {
-        assertEquals(Bar.class, mockingDetails(bar).getMockedType());
-    }
+    public void provides_invocations() {
+        //when
+        mock.simpleMethod(10);
+        mock.otherMethod();
 
-    @Test(expected = NotAMockException.class)
-    public void should_report_when_not_a_mockito_mock_on_getMockedType() throws Exception {
-        mockingDetails("not a mock").getMockedType();
+        //then
+        assertEquals(0, mockingDetails(foo).getInvocations().size());
+        assertEquals("[mock.simpleMethod(10);, mock.otherMethod();]", mockingDetails(mock).getInvocations().toString());
     }
 
     @Test
-    public void should_get_extra_interfaces() throws Exception {
-        Bar loup = mock(Bar.class, withSettings().extraInterfaces(List.class, Observer.class));
-        assertEquals(setOf(Observer.class, List.class), mockingDetails(loup).getExtraInterfaces());
+    public void provides_mock_creation_settings() {
+        //smoke test some creation settings
+        assertEquals(Foo.class, mockingDetails(foo).getMockCreationSettings().getTypeToMock());
+        assertEquals(Bar.class, mockingDetails(bar).getMockCreationSettings().getTypeToMock());
+        assertEquals(0, mockingDetails(mock).getMockCreationSettings().getExtraInterfaces().size());
     }
 
     @Test(expected = NotAMockException.class)
-    public void should_report_when_not_a_mockito_mock_on_getExtraInterfaces() throws Exception {
-        mockingDetails("not a mock").getExtraInterfaces();
+    public void fails_when_getting_creation_settings_for_incorrect_input() {
+        mockingDetails(null).getMockCreationSettings();
     }
 
-    private <T> Set<T> setOf(T... items) {
-        return new HashSet<T>(Arrays.asList(items));
+    @Test
+    public void fails_when_getting_invocations_when_null() {
+        try {
+            //when
+            mockingDetails(null).getInvocations();
+            //then
+            fail();
+        } catch (NotAMockException e) {
+            TestCase.assertEquals("Argument passed to Mockito.mockingDetails() should be a mock, but is null!", e.getMessage());
```

Comment:
@ChristianSchwarz I'm not a fan of `ExpectedException` rule, this make the test a bit awkward, try/catch is a bit more bulky but it offers more control.

Also when we switch to Java8 after mockito 2, we can bump to AssertJ 3.x which has interesting assertions like : 

```java
Assertions.assertThatThrownBy(() -> mockingDetails(null).getInvocations())
          .isInstanceOf(NotAMockException.class)
          .hasMessageContaining("Argument passed to Mockito.mockingDetails() should be a mock, but is null!")
```